File: source_code/constexpr_simple.cpp
/* Can be called in
run-time or compile-time */
static constexpr
int multiply(int a, int b) {
    return a * b;
}

/* Guarantee compile-time execution*/
static constexpr
    auto mult = multiply(1, 2);

int runtime_func(int a, int b) {
    /* Can also be
    used at run-time*/
    return multiply(a, b);
}

/* Can only be called
at compile-time */
static consteval
int multiply_c(int a, int b) {
    return a * b;
}

int runtime_func_c(int a, int b) {
    /* Will not compile, consteval
    function cannot be called at
    run-time*/
    return multiply_c(a, b);
}

File: source_code/constexpr_simple.rs
/* Can be called in
run-time or compile-time */
const fn
multiply(a: i32, b: i32) -> i32 {
    return a * b;
}

/* Guarantee compile-time
execution */
const mult: i32
    = multiply(1, 2);

pub fn
runtime_func(a: i32, b: i32) -> i32 {
    multiply(a, b);
}